C/services/south/ingest.cpp

```diff
@@ -499,14 +503,35 @@ void Ingest::processQueue()
 				m_failCnt = 0;
 				std::map<std::string, int>		statsEntriesCurrQueue;
 				AssetTracker *tracker = AssetTracker::getAssetTracker();
+				StorageAssetTracker *satracker = StorageAssetTracker::getStorageAssetTracker();
+				if ( satracker == nullptr)
+                                {
+                                        Logger::getLogger()->error("%s:%d could not initialize satracker ", __FILE__, __LINE__);
+					return;
+                                }
 
 				string lastAsset = "";
 				int *lastStat = NULL;
+				std::map <std::string , std::set<std::string> > assetDatapointMap;
+
 				for (vector<Reading *>::iterator it = q->begin();
 							 it != q->end(); ++it)
 				{
 					Reading *reading = *it;
 					string assetName = reading->getAssetName();
+                                        const std::vector<Datapoint *> dpVec = reading->getReadingData();
+					std::string temp;
+
+                                        for ( auto dp : dpVec)
+                                        {
+                                                temp.append("\"");
+                                                temp.append(dp->getName());
+                                                temp.append("\"");
+                                                assetDatapointMap[assetName].insert(temp);
```

Comment:
This insertion doesn't make sense in this loop. Please check.

Reply:
have modified data structure, its now map <asset , object> where object si a structure of { set<string> and count>

the set <string> will contain one reading dataPoints ( that too in sorted order)